frontend/src/pages/projects/screens/spawner/useNotebookSize.ts

```diff
@@ -28,15 +30,15 @@ export const useNotebookSize = (): {
   const setSelectedSizeSafe = React.useCallback(
     (name: string) => {
       let foundSize = sizes.find((size) => size.name === name);
-      if (!foundSize) {
+      if (!foundSize && isExistingNotebook) {
         [foundSize] = sizes;
         notification.warning(
           'The size you select is no longer available, we have set the size to the default one.',
         );
       }
-      setSelectedSize(foundSize);
+      setSelectedSize(foundSize ?? sizes[0]);
     },
-    [notification, sizes],
+    [notification, sizes, isExistingNotebook],
   );
```

Comment:
What's this logic doing? If it's a new notebook you're skipping the block anyways as there is no size selected 🤔 

Wouldn't we want to just do away with this callback?

Reply:
This logic is to prevent notifications from popping up in case of creating workbench. https://github.com/opendatahub-io/odh-dashboard/pull/2944#issuecomment-2205447763

Reply:
I understand what you've tried to do -- what I'm asking is why are you doing it and why are you duct taping the logic onto the side of the function and not just readdressing the function's purpose? You've effectively changed the logic to avoid a side-effect because your trying to avoid the side effect -- I'm asking you to spend time on understanding the code before just slapping a change to the logic. What's the use-cases here for this function now that use made this change?